### PR TITLE
Support inherited environment vars on exec

### PR DESCRIPTION
**Pull request by ulyssessouza**
**What I did**
Uses project's environment variables.

**Related issue**
Resolves https://github.com/docker/compose/issues/8852
Resolves https://github.com/docker/compose/issues/8746

### CODE REVIEW

File: pkg/compose/exec.go [hunk 1]
```diff
@@ -38,7 +39,7 @@ func (s *composeService) Exec(ctx context.Context, project string, opts api.RunO
 
 	exec, err := s.apiClient.ContainerExecCreate(ctx, container.ID, moby.ExecConfig{
 		Cmd:        opts.Command,
-		Env:        opts.Environment,
+		Env:        ToMobyEnv(types.NewMappingWithEquals(opts.Environment).Resolve(os.LookupEnv)),
```
Comment (ndeloof) [inline pkg/compose/exec.go:42]: Resolve should use project.Environment, not os.LookupEnv, so this also can resolve variables set by .env, and let us (in the future) get more control on the way environment is set

**Review by ndeloof — CHANGES_REQUESTED** [review]
LGTM but os.env vs project.env

File: pkg/compose/exec.go (hunk 1 above)
Comment (ulyssessouza) [reply pkg/compose/exec.go:42]: Hmmm... I think we need to change the interface for that. To construct the whole project we need to do that in the CLI side. So before arriving here.

File: cmd/compose/exec.go [hunk 2]
```diff
@@ -77,7 +78,7 @@ func execCommand(p *projectOptions, backend api.Service) *cobra.Command {
 }
 
 func runExec(ctx context.Context, backend api.Service, opts execOpts) error {
-	project, err := opts.toProjectName()
+	project, err := opts.toProject([]string{opts.service}, composegocli.WithResolvedPaths(true))
```
Comment (mix3) [inline cmd/compose/exec.go:81]: #8658 の修正とコンフリクトするように見えます
Project を渡す interface の修正に同意しますが、compose ファイルなしに Project を生成できる修正も同時に必要かもしれません
また、その修正は compose-spec/compose-go にも及ぶかもしれません

(DeepL) Looks like it conflicts with fix #8658
I agree with the fix for the interface that passes the Project, but there may also need to be a fix to generate the Project without the compose file.
Also, the fix may extend to compose-spec/compose-go

**Review by ndeloof — CHANGES_REQUESTED** [review]
that would be sad one can't `compose exec` without the original compose file (this is one of the major compose V2 improvements imho)
can't you pass environment to be applied to `Exec` by `api.RunOptions#Environment ` ?

**Comment by ulyssessouza** [comment]
@ndeloof Wow! I totally bypassed the fact that exec doesn't need the file anymore (V1 state of mind). Sorry!
Yep! I'll change that to get it from the options.
But actually looking at the code I just saw a a call to `toProjectName` [here](https://github.com/docker/compose/blob/v2/cmd/compose/exec.go#L80) and from what I have in mind, that will read the compose file at some point.

**Review by ndeloof — APPROVED** [review]

**Pull request merged**